deeplake/core/dataset/dataset.py

```diff
@@ -1767,6 +1768,35 @@ def _delete_branch(self, name: str) -> None:
             self._set_read_only(read_only, err=True)
             self.storage.autoflush = self._initial_autoflush.pop()
 
+    @invalid_view_op
+    def squash_commits(self) -> None:
```

Comment:
I left it as squash_commits() because while it's limited now, we can add behavior via arguments and/or removing limitations down the road and the method name continues to make sense.

For example, if/when we remove the limitation on only applying to to "main" it will be nice to not have a `squash_main` method hanging around to continue to support. The method can stay the same and users just get less validation errors when calling it from a non-main branch.

I did update the docstring slightly to not call out "it squashes main" but instead "it squashes the current branch" even if that current branch can only be main currently. 

That seem OK, @FayazRahman ?

Reply:
After further discussion, I added a commit for what had previously been committed and now has been squashed.